lib/webmock/util/headers.rb

```diff
@@ -6,18 +6,21 @@ module Util
 
     class Headers
 
+      STANDARD_HEADER_DELIMITER = '-'.freeze
+      NONSTANDARD_HEADER_DELIMITER = '_'.freeze
+      JOIN = ', '.freeze
+
       def self.normalize_headers(headers)
         return nil unless headers
-        array = headers.map { |name, value|
-          [name.to_s.split(/_|-/).map { |segment| segment.capitalize }.join("-"),
-           case value
+
+        headers.each_with_object({}) do |(name, value), new_headers|
```

Comment:
I'm not a maintainer, but `each_with_object` is an ActiveSupport method, and webmock doesn't depend on ActiveSupport. You'll probably need to change this for the test suite to pass.

Reply:
Wow. TIL. Thanks for the clarification!